swift/StableDiffusion/pipeline/StableDiffusionXLPipeline.swift

```diff
@@ -19,11 +19,14 @@ public struct StableDiffusionXLPipeline: StableDiffusionPipelineProtocol {
     public typealias Progress = PipelineProgress
     
     /// Model to generate embeddings for tokenized input text
-    var textEncoder: TextEncoderXLModel
+    var textEncoder: TextEncoderXLModel?
     var textEncoder2: TextEncoderXLModel
 
     /// Model used to predict noise residuals given an input, diffusion time step, and conditional embedding
     var unet: Unet
+    
+    /// Model used to refine the image, if present
+    var unetRefiner: Unet?
```

Comment:
Shall we make `unet` also an optional since we were talking about the following scenarios:
1-) Only refiner unet provided, we only refine
2-) Only base unet provided, we only run base gen
3-) Both provided, we run full pipeline (loading/unloading base/refiner during the transition)
4-) Neither provided, we error out

If you think (1-) is not relevant, we can discuss 👍

Reply:
Good question, 1-) might have a catch in the case that someone wants to use the refiner as a base model (in the case of `imageToImage` mode). If only a refiner is provided then we can assume it's `imageToImage` mode and require a `startingImage` as well. It might be useful to include some flexibility to use either a Unet or UnetRefiner for purely imageToImage applications to save disk space.

This question also goes back to torch2coreml.py I think - in the conversion script we could logically separate refiner unets in a similar way to how the controlnets are separate, and discourage using refiner unets as base models. Currently, refiner unets are considered the same as any other sdxl unet unless specifically set by the new argument. I do think there are cases where you'd want to use a "base" model as a refiner as more fine-tuned models/loras come out, and there are also some cases I've seen people use the refiner as a base model, but probably less common overall.

Reply:
I see! Let's keep it as is for now then. I will test the workflows as is.